### PR TITLE
adding tutorial exploring relationship between perturbation response and enrichment effect

### PR DESCRIPTION
**Pull request by ejiawustl**

### CODE REVIEW

**Comment by ejiawustl** [comment]
<p>Ok, I've added it as a dependency as it is what I am using to create the LOESS plots. </p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245467875'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Removed.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245467908'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Removed.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245467960'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Changed to show only the first 3 using this code. Also added another line to print the total number of TFs.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245467945'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Changed the description to focus on why we need this fn.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468025'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Added subheading.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468059'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Changed the description and added this to the docstring.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468093'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Updated the description to describe why we need to bin and moved part of the description to the docstring.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468129'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Updated the description and added typehinting and docstring.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468157'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Added both and updated the description above.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468188'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Updated the notebook to add subheadings that I thought were most appropriate.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468223'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Added analysis for this plot, and added explanation for the significance of the plot for our goal of identifying a general trend across TFs.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468255'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Added both for this method.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468280'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Commented this code out and only focused on 4 of these TFs. Provided some additional analysis below following the plots.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468301'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Added an explanation as to why we use these boxplots for further analysis.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468335'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>added typehinting and docstring to the method.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468375'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Modified the method to remove unnecessary lines and show only the boxplot.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468405'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Added analysis to explain the significance of these plots.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245468441'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by cmatKhan** [comment]
<p>It is not part of this commit -- it hasn't been added</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245467875'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ejiawustl** [comment]
<p>Oops you are right. I originally added it in a different cell but it gave me an error while testing other things so I removed it and forgot to add it back in. I have noted this and will update this along with anything else in the next commit.</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245467875'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: docs/tutorials/exploring_perturbation_response_relationship.ipynb [hunk 1]
```diff
@@ -0,0 +1,1053 @@
+{
```
Comment (cmatKhan) [inline docs/tutorials/exploring_perturbation_response_relationship.ipynb:1]: <p>Make the main header match the filename, in this case</p><p> </p><h1>Exploring the Binding/Perturbation Relationship</h1><p> </p><p>the filename should also be exploring_binding_perturbation relationship.</p><p> </p><p>I realize i said perturbation response, but that is not as clear as binding perturbation relationship</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (cmatKhan) [inline docs/tutorials/exploring_perturbation_response_relationship.ipynb:1]: <p>This is a docstring. If you are including text, then make the text expository. Explain why you need the function. docstring documentation is for developers. The tutorial is an example of using the package</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (cmatKhan) [inline docs/tutorials/exploring_perturbation_response_relationship.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.        Params:
</pre><p>the docstring style of this entire package is sphinx</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (cmatKhan) [inline docs/tutorials/exploring_perturbation_response_relationship.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.        Params:
</pre><p>package docstring format is sphinx</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (cmatKhan) [inline docs/tutorials/exploring_perturbation_response_relationship.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.        Params:
</pre><p>package docstring format is sphinx</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (cmatKhan) [inline docs/tutorials/exploring_perturbation_response_relationship.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #6.        Params:
</pre><p>the package docstring style is sphinx</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (cmatKhan) [inline docs/tutorials/exploring_perturbation_response_relationship.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #8.        Params:
</pre><p>docstring style of the package is sphinx</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

**Comment by cmatKhan** [comment]
<p>You didn't push the commit after adding this -- this isn't resolved yet</p><p> </p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245467875'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by cmatKhan** [comment]
<p>It doesn't appear that you pushed the commit that includes this change</p>

---

 <a href='https://app.reviewnb.com/BrentLab/yeastdnnexplorer/pull/94/discussion/#comment-2245467993'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by cmatKhan** [comment]
closed in order to add conclusion

**Pull request closed without merging**